compiler/src/vm/builtin_functions.rs

```diff
@@ -78,6 +78,27 @@ impl From<SuccessfulBehavior> for BuiltinResult {
     }
 }
 
+macro_rules! unpack {
+    ( $heap:expr, $args:expr, ($( $arg:ident: $type:ty ),+), $body:block ) => {
+        {
+            let ( $( $arg, )+ ) = if let [$( $arg, )+] = $args {
+                ( $( *$arg, )+ )
+            } else {
+                return Err(
+                    "a builtin function was called with the wrong number of arguments".to_string(),
+                );
+            };
+            let ( $( $arg, )+ ): ( $( UnpackedData<$type>, )+ ) = ( $(
+                UnpackedData {
+                    address: $arg,
+                    data: $heap.get($arg).data.clone().try_into()?,
+                },
+            )+ );
+
+            $body.into()
```

Comment:
Can we reuse this in `unpack_and_later_drop!(…)`?

Reply:
For some reason, this doesn't work.
In particular, this causes variables in arguments to not be resolved.

![image](https://user-images.githubusercontent.com/8601189/186892034-e43ac1ef-0ae8-4d09-8828-ab4d8dbfd0a4.png)